detectors/azure/vm/vm.go

```diff
@@ -0,0 +1,137 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package vm // import "go.opentelemetry.io/contrib/detectors/azure/vm"
```

Comment:
`vm` doesn't seem like the best package name.  consider `azurevm`, or something that is likely to be relatively unique.

Reply:
Changed it to `azurevm`.

I went with `vm` initially, as `detectors/azure/azurevm` seemed a bit redundant.